k8s.io/test.py

```diff
@@ -39,194 +39,220 @@ def do_get(url):
 
 
 class RedirTest(unittest.TestCase):
-    def assert_redirect(self, partial_url, expected_loc, **kwargs):
+    def assert_code(self, url, expected_code):
+        print('GET: %s => %s' % (url, expected_code))
+        resp, body = do_get(url)
+        self.assertEqual(resp.status, expected_code)
+
+    def assert_scheme_redirect(self, url, expected_loc, expected_code, **kwargs):
+        for k, v in kwargs.items():
+            k = '$%s' % k
+            v = '%s' % v
+            url = url.replace(k, v)
+            expected_loc = expected_loc.replace(k, v)
+        print('REDIR: %s => %s' % (url, expected_loc))
+        resp, body = do_get(url)
+        self.assertEqual(resp.status, expected_code)
+        self.assertEqual(resp.getheader('location'), expected_loc)
+
+    def assert_redirect(self, partial_url, expected_loc, expected_code, **kwargs):
```

Comment:
since `assert_perm_redirect` is only used twice, it would be cleaner to just make expected_code a default arg:

    def assert_redirect(self, partial_url, expected_loc, expected_code=302, **kwargs)